packages/core/src/components/dialog/_dialog.scss

```diff
@@ -106,10 +106,17 @@ $dialog-header-padding: math.div($pt-grid-size, 2);
 
   .#{$ns}-icon-large,
   .#{$ns}-icon {
-    color: $pt-icon-color;
     flex: 0 0 auto;
     margin-left: -3px;
     margin-right: $dialog-padding * 0.5;
+
+    // only apply light color to icons that are not intent-specific
+    :not([class*="#{$ns}-intent"]) {
+      &.#{$ns}-icon-large,
+      &.#{$ns}-icon {
```

Comment:
These shouldn't be necessary either (except maybe for specificity); you're assembling a selector like `.#{$ns}-icon:not([class*="#{$ns}-intent"]).#{$ns}-icon`, so you have the icon class in there twice.

Reply:
🤦‍♂️ Not quite sure how I missed that one.. Thanks for the catch! 2d4ae487a3c6b54ca4894576e91b34a0218c39b0